### PR TITLE
Add json output

### PR DESCRIPTION
**Pull request by bhagenbourger**
#32

### CODE REVIEW

**Comment by bhagenbourger** [comment]
Hello @vianneybacoup ,
In this PR I put all test results into "target" folder excepted "output.*" because it's the default value for the output_name.
I do that to keep root folder "clean", to easier clean test results and also to avoid to add all extensions into the .gitignore file.
But, as default value is "output", I don't really know if it's really useful.
What do you think about that?

Thank you for your feedback.

**Comment by vianneybacoup** [comment]
> Hello @vianneybacoup , In this PR I put all test results into "target" folder excepted "output.*" because it's the default value for the output_name. I do that to keep root folder "clean", to easier clean test results and also to avoid to add all extensions into the .gitignore file. But, as default value is "output", I don't really know if it's really useful. What do you think about that?
> 
> Thank you for your feedback.

I see that we both had the same issues, I did way too much 'rm *.parquet *.csv' past months haha
I had in mind we could have like in Python a tearDown fucntion that cleanup the output of the test automatically, but that does not exist in rust, and this is not a mandatory feature, the folder is clearly enough.
Maybe rename it with target/test_generated or something like this just to avoid a possible confusion with the other targets?

**Comment by bhagenbourger** [comment]
I moved output tests into `target/test_generated/` folder.
I kept "all options" tests into `target` folder because they are run only by github action and generation fails if folder not exists (maybe could be a good enhancement to automatically generate intermediate folders?).
I added `ctor` crate to add macro to create `target/test_generated/` folder before tests and clean test outputs after tests.

About json output, I did a POC using arrow (https://github.com/bhagenbourger/Fakelake/tree/poc/use_arrow_for_all_format) because I found interesting to use the same "generator" for all formats. It works but with some limitations as using the same date format for all columns. So definitively, serde-json is more flexible and I keep this implementation.

**Review by vianneybacoup — CHANGES_REQUESTED** [review]
About arrow for json, I am also wondering if it has poorer performances than serde-json.
Fakelake whole purpose is the speed, so I'd say it should be the focus

Small adjustements to be done for this PR, thank you for the work :)

File: tests/json_all_options.yaml [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+  - name: score
+    provider: Random.Number.i32
+    min: -100
+    max: 100
+
+  - name: percentage
+    provider: Random.Number.f64
+    min: -1000
+    max: 1000
+
```
[the remark this answers is not in the document]
Comment (bhagenbourger) [reply tests/json_all_options.yaml:49]: I forgot to do that after the rebase, fixed.

**Review by vianneybacoup — APPROVED** [review]
Thanks ! :)

**Pull request merged**